pegasus/sites.v3/code.org/public/educate/weblab.html

```diff
@@ -0,0 +1,222 @@
+          successful.
+        </p>
+        <label for="connect"
+          >Click "Connect" to check whether your school's network supports the
+          Web Lab tool</label
+        >
+        <div id="status-table">
+          <table>
+            <tbody>
+              <tr>
+                <th>Check</th>
+                <th>Status</th>
+              </tr>
+              <tr>
+                <td>
+                  Connected to
+                  <code>https://downloads.computinginthecore.org</code>
+                </td>
+                <td class="computinginthecore">
+                  <p class="incomplete">Not complete</p>
+                  <p class="loading" style="display: none">Connecting...</p>
+                  <p class="succeeded" style="display: none">Success</p>
+                  <p class="failed" style="display: none">Failed</p>
+                </td>
+              </tr>
+              <tr>
+                <td>Connected to <code>https://codeprojects.org</code></td>
+                <td class="codeprojects">
+                  <p class="incomplete">Not complete</p>
+                  <p class="loading" style="display: none">Connecting...</p>
+                  <p class="succeeded" style="display: none">Success</p>
+                  <p class="failed" style="display: none">Failed</p>
+                </td>
+              </tr>
+              <tr>
+                <td>Able to load an iFrame</td>
+                <td class="iframe">
+                  <p class="incomplete">Not complete</p>
+                  <p class="loading" style="display: none">Connecting...</p>
+                  <p class="succeeded" style="display: none">Success</p>
+                  <p class="failed" style="display: none">Failed</p>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+        <div>
+          <button id="connect" class="btn btn-primary">Connect</button>
+        </div>
+        <p id="success-message" style="display: none">
+          <strong>You passed all checks! No further action is needed.</strong>
+        </p>
+        <div id="fail-message" style="display: none">
+          <p>
+            <strong>Error. One or more checks failed.</strong>
+          </p>
+          <p>
+            <strong>Action required: </strong>
+            Please contact your IT department to update your school’s firewall
+            settings or device policies. Here is the rest of this call to
+            action.
+          </p>
+          <p>
+            Contact our Support Team (<a href="mailto:support@code.org"
+              >support@code.org</a
+            >) if you have any questions.
+          </p>
+        </div>
+        <iframe id="test-iframe" src="/" style="display: none"></iframe>
+      </div>
+    </div>
+  </div>
+</div>
+<script>
+  // TODO: Is there a way to use the existing code for this in apps directory?
```

Comment:
In fact, we shouldn't have JS in this file, and rather all of this JS should come out of apps/ so that it goes through our full build process.  Then you can also use shared components, and in the future we can maybe use some of this code in other places too.